### PR TITLE
(#389)Creating-A-Video

### LINKED ISSUE #389: CMS: Create a video
## As a user I want to be able to add a new video 

> **ESTIMATE** TBD

### Acceptance Criteria

```gherkin
 Scenario: create a new video
        Given user is navigating to "/user/login"
        When user enters credentials
        And user clicks "Log in" button
        Then user is logged in and the user name "admin" is displayed in the toolbar
        And the tool bar appears at the top
        When user clicks on "Content" tab
        And user clicks on "Media" sub tab
        And user clicks on "Add media" action button
        And user clicks on "Video" content type
        Then page title is "Add Video"
        When user selects test site section
        And user fills out the following fields
            | fieldLabel        | value                                       | field_name                         |
            | Youtube Video URL | https://www.youtube.com/watch?v=zXE7rexUCeI | field_media_oembed_video[0][value] |
            | Pretty URL        | test-video                                  | field_pretty_url                   |
            | Name              | Test Video                                  | name[0][value]                     |
            | Browser Title     | Test Video                                  | field_browser_title                |
            | Meta Description  | Test Video - Meta Description               | field_page_description             |
        And user types "Test Caption" into Caption text field
           And user selects "Published" from "Save as" dropdown
        Then user saves the content page
       


    Scenario: Create test pages to test video
        Given user is navigating to "/user/login"
        When user enters credentials
        And user clicks "Log in" button
        Then user is logged in and the user name "admin" is displayed in the toolbar
        And the tool bar appears at the top
        When user clicks on "Content" tab
        And user clicks on "Add content" action button
        And user clicks on "Home and Landing" content type
        Then page title is "Home and Landing"
        When user selects test site section
        And user fills out the following fields
            | fieldLabel       | value                                             | field_name             |
            | Pretty URL       | home-and-landing-page-video                       | field_pretty_url       |
            | Page Title       | Home and Landing page for Video                   | title                  |
            | Browser Title    | Home and Landing page for Video - Browser Title  | field_browser_title    |
            | Meta Description | Home and Landing page for ImVideoage  Meta Description | field_page_description |
        And user selects "Add Multimedia Row" from "Contents" dropdown
        And browser waits
        And user clicks on "Multimedia Card" link in the "Multimedia Row" text area
        And browser waits
        And user clicks on "Select media" button item
        And user selects "Test Video" from the list of media
        And user clicks on "Select media" button to select media
        And browser waits
        And user selects "Published" from Save as dropdown
        When user saves the content page

    Scenario: Verify video 
        When user is navigating to the front end site with path site section plus "home-and-landing-page-video"
        Then the "Test Video" titles video is displayed
```

### Resources:
* Feature: JIRA Ticket #
* Mockups: *Here goes a URL to or the name of the mockup(s) in inVision*
* Web Page: *Here goes a URL to an existing web page*

### Notes

*Some complementary notes if necessary:*

* > Here goes a quote from an email
* Here goes whatever useful information can exist…

---

### Prerequisites
- *required dependency for identifying the problem, scope and then solution for this issue*
- [ ] Short Spike Description - Issue #9999

### Tasks
- [ ] Short Task Description - Issue #9999
- [ ] Short Task Description - Issue #9999

### Technical Notes
- More notes, or technical nuances/direction

### PR DESCRIPTION
**Pull request by ahmadakbaryNIH**
closes #389

### CODE REVIEW

**Comment by ahmadakbaryNIH** [comment]
CMS: Creating Video has been completely Automated at dev-ac and cm-test environment.
image
image

File: cypress/e2e/CMS/$CreatingAvideo.feature [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    And user clicks on "Multimedia Card" link in the "Multimedia Row" text area
+    And browser waits
+    And user clicks on "Select media" button item
+    And browser waits
+    And user selects "Test Video" from the list of media
+    And browser waits
+    And user clicks on "Select media" button to select media
+    And browser waits
+    And user selects "Published" from "Save as" dropdown
+    When user saves the content page
```
Comment (KateMashkinaNIH) [inline cypress/e2e/CMS/$CreatingAvideo.feature:58]: so this scenario has 5 waits added - that means extra 10 second just for this test to execute on top of the rest of the steps. Do you need all of the waits, e.g. what happens if you remove them?

File: cypress/e2e/CMS/$CreatingAvideo/index.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+And('user selects a checkbox next to title with url {string} from the list of content', (url) => {
+    cy.get(`a[href='${siteSection}/${url}-${randomNum}']`).parent().parent().find('input.form-checkbox').check();
+});
+
+Given('user is navigating to the front end site with path site section plus {string}', (purl) => {
+    cy.visit(`${frontEndBaseUrl}${siteSection}/${purl}-${randomNum}`, { retryOnStatusCodeFailure: true });
+});
+
+And('the content item with url {string} does not exist in the list of content', (url) => {
+    cy.get(`a[href='${siteSection}/${url}-${randomNum}']`).should('not.exist');
```
Comment (KateMashkinaNIH) [inline cypress/e2e/CMS/$CreatingAvideo/index.js:39]: also, why do you need a random url steps here? You only create page once and there is only one section added- remove it and it should work

File: cypress/e2e/CMS/$CreatingAvideo/index.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+});
+
+And('user types {string} into Caption text field', (value) => {
+    cy.getNthIframe("iframe[class='cke_wysiwyg_frame cke_reset']", 1).find('p').type(value)
+})
+
+And('user selects {string} from {string} dropdown', (dropdwon, section) => {
```
Comment (KateMashkinaNIH) [inline cypress/e2e/CMS/$CreatingAvideo/index.js:46]: we have this step implemented in common - why did you need to duplicate it here?

**Review by KateMashkinaNIH — CHANGES_REQUESTED** [review]

File: cypress/e2e/CMS/$CreatingAvideo.feature (hunk 1 above)
Comment (ahmadakbaryNIH) [reply cypress/e2e/CMS/$CreatingAvideo.feature:58]: Removed

File: cypress/e2e/CMS/$CreatingAvideo/index.js (hunk 2 above)
Comment (ahmadakbaryNIH) [reply cypress/e2e/CMS/$CreatingAvideo/index.js:39]: Removed

File: cypress/e2e/CMS/$CreatingAvideo/index.js (hunk 3 above)
Comment (ahmadakbaryNIH) [reply cypress/e2e/CMS/$CreatingAvideo/index.js:46]: Removed duplicated step!

Comment (ahmadakbaryNIH) [reply cypress/e2e/CMS/$CreatingAvideo/index.js]: image

**Review by KateMashkinaNIH — APPROVED** [review]
